### PR TITLE
Transactions worker

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

A Web Worker that periodically checks for new ICRC transactions.

# Notes

This introduces the worker but, not its usage.

# PRs

- [x] need #2754

# Changes

- new `transactions.worker`
- related `initTransactionsWorker` service which can be use to load and interact with the worker in a component

### CODE REVIEW

File: frontend/src/lib/constants/accounts.constants.ts [hunk 1]
```diff
@@ -1,3 +1,8 @@
+import { SECONDS_IN_MINUTE } from "$lib/constants/constants";
+
 export const HARDWARE_WALLET_NAME_MIN_LENGTH = 2;
 export const SYNC_ACCOUNTS_RETRY_SECONDS = 2;
 export const SYNC_ACCOUNTS_RETRY_MAX_ATTEMPTS = 50;
+
+// Workers
+export const SYNC_ACCOUNTS_TIMER_INTERVAL = (SECONDS_IN_MINUTE / 2) * 1000; // 30 seconds in milliseconds
```
Comment (lmuntaner) [inline frontend/src/lib/constants/accounts.constants.ts:8]: Shall we add the unit to the name? `SYNC..._MILLIS`?

File: frontend/src/tests/lib/services/worker-transactions.services.spec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          msg: "nnsStartTransactionsTimer";
+          data: PostMessageDataRequestTransactions;
+        }) {
+          spyPostMessage(data);
+        }
+      };
+    });
+  });
+
+  it("should start worker with params", async () => {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/services/worker-transactions.services.spec.ts:25]: Can you check also the `stopTransactionsTimer`?

**Review by lmuntaner — APPROVED** [review]
LGTM, thanks!

File: frontend/src/lib/constants/accounts.constants.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/constants/accounts.constants.ts]: Sure

File: frontend/src/tests/lib/services/worker-transactions.services.spec.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/tests/lib/services/worker-transactions.services.spec.ts:25]: Absolutely

**Pull request merged**